### ISSUE TITLE
User gets registered in database before verification.

### ISSUE THREAD

**Issue by DextersCode** [foo290/Django-Verify-Email#32]
Django-Verify-Email is awesome but I have an issue.
The problem is that, when the user clicks on register, before verifying through gmail, the user gets registered in database as a non-active user.
If not verified, the user cannot login, which is good but if user can't verify for some reason and tries to register again, the username shows 'taken'. So is there any solution for this, which I'm not aware of ?? 
In other words how would I make them retry for verification, even if the user is registered ??

**Comment by foo290** [comment]
well, prevention from registering using the same username again is well desired in almost every platform as the username must be unique, if a user forgets/deletes the email from the first try, u can implement some portal on the frontend side for re-requesting the link, the logic would be:

check if the user requesting is inactive and the last login is not been attempted yet, if so, then generate and send the link again to his email (or some similar logic), the point is u need to do most of the part as most of the work lands on your preference of making re-send logic and making frontend for that.

or maybe I'd do something about this in the future in my free time :)

**Comment by DextersCode** [comment]
Thanks for replying that quick :)
> check if the user requesting is inactive and the last login is not been attempted yet, if so, then generate and send the link again to his email (or some similar logic), the point is u need to do most of the part as most of the work lands on your preference of making re-send logic and making frontend for that.

well I thought the same, I've created a new page view to retry the request and created the same form asking for email, it does create a new token and sends a new email. But it also creates a new entry in dataset, so there are two entries in database with same email. It does not update the previous one. So things were getting way complicated and out of hands.

So if there's any way you could provide a retry template like VERIFICATION_RETRY_TEMPLATE or something like that, that would be very much helpful.
Also I noticed that there is an Expiry template after the token gets expired if I update the EXPIRE_AFTER variable. Again that would be a great help if there's a variable to add a custom html template for that.

Thanks for creating this library, it helped a lot ♡.

**Comment by foo290** [comment]
yea I have the logic and code in mind will do it on weekend maybe release it next week, thanks for using giving valuable new feature insights :)

**Comment by foo290** [comment]
`Also I noticed that there is an Expiry template after the token gets expired if I update the EXPIRE_AFTER variable. Again that would be a great help if there's a variable to add a custom html template for that.`

there is already a variable for that `LINK_EXPIRED_TEMPLATE`

changes will be live on pypi in next 30 mins from now, thanks!

**Comment by DextersCode** [comment]
Wow Thankyou verymuch, I'll try it and let you know 😃

**Comment by DextersCode** [comment]
Hey it works, thanks for the update!
Also there is a typo in documentation, which doesn't matter that much, but letting you know.

![Screenshot (395)](https://user-images.githubusercontent.com/70622181/158050078-06c5f8d9-5c76-484b-b812-b3aadc456fcd.png)
